src/core/qgsgpkgflusher.cpp

```diff
@@ -28,13 +28,20 @@ class Flusher : public QObject
 {
     Q_OBJECT
 
+  public:
+    static QMutex sMutex;
```

Comment:
This could be private non static I think

Reply:
Yes, it can. I made it public static before I realized there is no need for lock from outside (`QgsGpkgFlusher::stop`).